GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+--runtime-fetch-client-parallel           If defer hooks should be started at the same time as the blocking ones.
+```
+
+**Important:** Remember that a component must be mapped in the route definition for the hooks on it to be executed.
+
+## Data flow
+[Redux](http://redux.js.org/docs/introduction/) is a very popular library that implements unidirectional flow of data along with a great way of `connect`ing this store of data to any of your React components.
+
+As our applications grow in size and complexity; handling `state` using just React `Component` and `setState` can start to become tedious.
+
+`web-app-react` integrates `Redux` as a first-class citizen and it is available for you to use. Roc does not modify the API of Redux; it only aims to be readily configured.
+
+### Connecting components to the Redux store
+
+A connected component will automatically be updated when the Redux state data is modified.  
+
+Example of a redux-connected component in a project:
+
+```javascript
+import { connect } from 'react-redux';
+
+@connect(state => ({
+  myprop: state.somevalue,
+}))
+export default class MyComponent extends Component {
+  static propTypes = {
+    myprop: React.PropTypes.number,
+  }
+
+  render() {
+    const { myprop } = this.props;
+
+    return (
+      <h1>Redux demo: { myprop }</h1>
+    );
+  }
+}
+```
+
+The value from Redux `state.somevalue` is now connected to `MyComponent` accessible as the `prop`: `myprop`.
+
+### Redux actions
+
+#### Defining actions
+Roc does not dictate how you structure your actions or action creators. You just need to be able to import them in required contexts.
+
+Example `actions.js`:
+
+```javascript
+export function myAction() {
+  return {
+    type: 'SOMETHING'
+  };
+}
+```
+
+#### Dispatching actions
+Roc does not dictate how you dispatch actions. Just import your action and `dispatch` it. `dispatch` is available in the object passed as argument to the data fetching hooks and as a `prop` on Redux `connect`ed React components.
+
+Example connected component with `dispatch` as a prop:
+```javascript
+import { connect } from 'react-redux';
+import { myAction } from './actions';
+
+@connect()
+class ConnectedComponent extends Component {
+  render() {
+    return (
+      <button onClick={ () => { this.props.dispatch(myAction()); } } />
+    );
+  }
+}
+```
+
+Example component with hooks that uses `dispatch`:
+
+```javascript
+import { connect } from 'react-redux';
+import { myAction } from './actions';
+
+@provideHooks({
+  fetch: ({ dispatch }) => dispatch(myAction()).then(/*something...*/),
+})
+class HookComponent extends Component {
+  render() {
+    return (
+      <div />
+    );
+  }
+}
+```
+
+If you answered `Y` to include a Redux demo in your app you can view some example actions in the bottom of `src/redux/repo.js`
+
+### Redux reducers
+Roc bootstraps Redux for us and expects to find your reducers exported in the file configured by
```

Comment:
Should this be higher? Without a reducers file we don't get Redux in the application.